piecrust/src/session/call_stack.rs

```diff
@@ -65,36 +64,14 @@ impl CallStack {
             panic!("Module not in the stack: {module_id:?}");
         }
 
-        let (_, count) = self.instance_map.get_mut(&module_id).unwrap();
-        *count += 1;
-
         self.stack.push(StackElement { module_id, limit });
     }
 
     /// Pops an element from the callstack.
     ///
     /// If it is a the last element in the stack for the popped instance.
     pub fn pop(&mut self) {
-        if let Some(element) = self.stack.pop() {
-            let mut entry = match self.instance_map.entry(element.module_id) {
-                Entry::Occupied(e) => e,
-                _ => unreachable!("map must have an instance here"),
-            };
-
-            let (instance, count) = entry.get_mut();
-            *count -= 1;
-
-            if *count == 0 {
-                // SAFETY: This is the last instance of the module in the
-                // stack, therefore we should recoup the memory and drop
-                //
-                // Any pointers to it will be left dangling
-                unsafe {
-                    let _ = Box::from_raw(*instance);
-                    entry.remove();
-                };
```

Comment:
If an instance is never removed from the map and dropped, then the linear memory `mmap` it is holding is never reloaded from disk - this is why the crossover test is failing